### PR TITLE
Waste less space in the main Transaction type

### PR DESCRIPTION
**Pull request by jannotti**
Use a pointer to HeartbeatTxnFields instead of embedded the entire, large structure.

### CODE REVIEW

File: data/transactions/transaction.go [hunk 1]
```diff
@@ -632,11 +636,11 @@ func (tx Transaction) WellFormed(spec SpecialAddresses, proto config.ConsensusPa
 		nonZeroFields[protocol.ApplicationCallTx] = true
 	}
 
-	if !tx.StateProofTxnFields.Empty() {
+	if !tx.StateProofTxnFields.MsgIsZero() {
```
Comment (jannotti) [inline data/transactions/transaction.go:639]: We don't need this change, of course, I just noticed that `Empty()` is just a re-implementation of `MsgIsZero`. We should either revert this line, or remove `Empty()` I suppose.

Comment (gmalouf) [reply data/transactions/transaction.go:639]: I think fine to take this.

File: tools/x-repo-types/typeAnalyzer/typeAnalyzer.go [hunk 2]
```diff
@@ -237,12 +237,17 @@ func (t *TypeNode) buildStructChildren(path TypePath) TypePath {
 
 		if typeField.Anonymous {
 			// embedded struct case
-			actualKind := typeField.Type.Kind()
+			fieldType := typeField.Type
```
Comment (gmalouf) [inline tools/x-repo-types/typeAnalyzer/typeAnalyzer.go:240]: Assume this was to handle if we don't use pointer in SDK, fine with me.

File: util/set.go [hunk 3]
```diff
@@ -54,7 +54,9 @@ func Union[T comparable](sets ...Set[T]) Set[T] {
 }
 
 // Intersection constructs a new set, containing all elements that appear in all
-// given sets. nil is never returned.
+// given sets. nil is never returned. Intersection of no sets is an empty set
+// because that seems more useful, regardless of your very reasonable arguments
```
Comment (gmalouf) [inline util/set.go:58]: (crying)

**Review by gmalouf — APPROVED** [review]

**Pull request merged**